cmd/syncthing/gui.go

```diff
@@ -1567,14 +1609,38 @@ func browseFiles(current string, fsType fs.FilesystemType) []string {
 
 	fs := fs.NewFilesystem(fsType, searchDir)
 
-	subdirectories, _ := fs.Glob(searchFile + "*")
+	subdirectories, _ := fs.DirNames(".")
+
+	matches := make([]match, 0, len(subdirectories))
 
-	ret := make([]string, 0, len(subdirectories))
 	for _, subdirectory := range subdirectories {
 		info, err := fs.Stat(subdirectory)
-		if err == nil && info.IsDir() {
-			ret = append(ret, filepath.Join(searchDir, subdirectory)+pathSeparator)
+		if !(err == nil && info.IsDir()) {
+			continue
+		}
+
+		m := match{
+			path:            filepath.Join(searchDir, subdirectory) + pathSeparator,
+			caseDifferences: prefixCaseDifferences(subdirectory, searchFile),
 		}
+
+		if m.caseDifferences >= 0 {
+			matches = append(matches, m)
```

Comment:
You could store the prefifDiff on the stack, and if that is not negative, only then create match in the append statement.